### ISSUE TITLE
Migrate to gRPC to talk to LND directly

### ISSUE THREAD

**Issue by AnotherDroog** [lncm/invoicer-ui#6]
Does this require nginx reverse proxy?

**Comment by hitchcott** [comment]
Not a requirement, but need to figure out how to provision a macroon and hand credentials to the client. 

Also, gRPC only works over HTTPS (so on a local network you need to send a self-signed cert to the client too, afaik). 

REST is also an option.

**Comment by AnotherDroog** [comment]
Okay this will need more thought.

gRPC requires HTTPS, even when connecting locally?

I can investigate let’s encrypt certificates for local hostnames again, Damian found a trick there.

Otherwise, REST was nice to work with, as long as we can receive payment status on client side

**Comment by hitchcott** [comment]
> gRPC requires HTTPS, even when connecting locally?

Yes, but depending on how strict the browser transport policy is, we might be able to use a self-signed certificate; we could serve the app over non-encrypted HTTP, and use gRPC with a self-signed HTTPS cert.

Would need to research.

https://github.com/grpc/grpc-web

**Comment by nolim1t** [comment]
We can indeed get nginx happening assuming we can get a dockerfile and container for it

https://github.com/lncm/dockerfiles/issues/17
